chain/cmd/bandsv/main.go

```diff
@@ -26,14 +26,31 @@ import (
 	"github.com/bandprotocol/d3n/chain/wasm"
 )
 
+const (
+	Asynchronous = "ASYNCHRONOUS"
+	Synchronous  = "SYNCHRONOUS"
+	Full         = "FULL"
+)
+
 type OracleRequest struct {
+	Type     string          `json:"type" binding:"required"`
 	CodeHash cmn.HexBytes    `json:"codeHash" binding:"len=32"`
 	Params   json.RawMessage `json:"params" binding:"required"`
 }
 
-type OracleRequestResp struct {
-	RequestId uint64 `json:"id"`
+type OracleRequestAsynchronousResp struct {
+	TxHash string `json:"txHash"`
+}
+
+type OracleRequestSynchronousResp struct {
 	TxHash    string `json:"txHash"`
+	RequestId uint64 `json:"id"`
+}
+
+type OracleRequestFullResp struct {
```

Comment:
I think we can combine the struct. 
```
type OracleRequestResp struct {
    TxHash string
    RequestId *uint64
    Proof ...
}
```
Then just set to nil if not applicable. Would that work @taobun ?